core/src/main/java/org/apache/cxf/common/util/PackageUtils.java

```diff
@@ -56,7 +57,9 @@ public static String getSharedPackageName(List<Class<?>> classes) {
         List<List<String>> lParts = new ArrayList<List<String>>(classes.size());
         List<String> currentParts = new ArrayList<>();
         for (Class<?> cls : classes) {
-            lParts.add(StringUtils.getParts(getPackageName(cls), "\\."));
+            if (!Proxy.isProxyClass(cls)) {
```

Comment:
Good idea, I first thought to use InvocationHandler stuff for this but in the method only the class itself is available. And afaik it's not possible to get the real class as there might be multiple interfaces...